### ISSUE TITLE
[Bug] [daily] UNION ALL query return wrong 

### ISSUE THREAD

**Issue by zhangy5** [apache/doris#16505]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master: doris-0.0.0-trunk-cf18de1

### What's Wrong?

jenkins failed case: https://ci-builds.apache.org/job/Doris/job/doris_daily_enable_vectorized/201/testReport/(root)/test_execute/test_issue_4772/

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/10826066/217445654-2d85853a-514a-4f63-9f6d-6d076f24655a.png">

https://github.com/apache/doris/issues/4772

### What You Expected?

return 2 rows

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by luozenglin** [comment]
This problem is caused by the constant conjunctions evaluating the empty result set incorrectly:
```
mysql> select * from (select k1, -1 as c from test_query_db.test) t where t.c = 1;
+------+------+
| k1   | c    |
+------+------+
|    3 |   -1 |
|    1 |   -1 |
|    2 |   -1 |
+------+------+
```

Expected:
```
mysql> select * from (select k1, -1 as c from test_query_db.test) t where t.c = 1;
Empty set (0.04 sec)
```

**Comment by luozenglin** [comment]
fixed by #17036
